pkg/nsx/services/vpc/builder.go

```diff
@@ -64,9 +64,9 @@ func buildNSXVPC(obj *v1alpha1.NetworkInfo, nsObj *v1.Namespace, nc common.VPCNe
 		vpc = nsxVPC
 	} else {
 		// for creating vpc case, fill in vpc properties based on networkconfig
-		vpcName := util.GenerateDisplayName("", "vpc", obj.GetNamespace(), "", cluster)
+		vpcName := util.GenerateTruncName(common.MaxNameLength, obj.GetName(), "", "", "", cluster)
 		vpc.DisplayName = &vpcName
-		vpc.Id = common.String(string(nsObj.GetUID()))
+		vpc.Id = common.String(util.GenerateIDByObject(obj))
```

Comment:
We also need to check if VPC LBS id should be changed. See #618 

Note: VPC LBS will be consumed by NCP.